sbol_utilities/component.py

```diff
@@ -621,4 +622,84 @@ def part_in_backbone(identity: str, part: sbol3.Component, backbone: sbol3.Compo
     part_in_backbone_component.features.append(backbone_subcomponent)
     # adding topology
     part_in_backbone_component.types.append(topology_type)
-    return part_in_backbone_component, part_in_backbone_seq
\ No newline at end of file
+    return part_in_backbone_component, part_in_backbone_seq
+
+def digestion(reactant:sbol3.Component, restriction_enzymes:List[sbol3.ExternallyDefined], assembly_plan:sbol3.Component)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """Digests a Component using the provided restriction enzymes and creates a product Component and a digestion Interaction.
+
+    :param reactant: DNA to be digested as SBOL Component. 
+    :param restriction_enzymes: Restriction enzymes used  Externally Defined.
+    :return: A tuple of Component and Interaction.
+    """
+    if all(t != sbol3.SBO_DNA for t in reactant.types):
+        raise TypeError(f'The reactant should has a DNA type. Types founded {reactant.types}.')
+    if len(reactant.sequences)!=1:
+        raise ValueError(f'The reactant needs to have only one sequence. The input reactant has {len(reactant.sequences)} sequences')
+    participations=[]
+    restriction_enzymes_pydna=[] 
+    for re in restriction_enzymes:
+        enzyme = Restriction.__dict__[re.name]
+        restriction_enzymes_pydna.append(enzyme)
+        assembly_plan.features.append(re)
+        modifier_participation = sbol3.Participation(roles=[sbol3.SBO_MODIFIER], participant=re)
+        participations.append(modifier_participation)
+
+    # Inform topology to PyDNA, if not found assuming circular. 
+    if is_circular(reactant):
+        circular=True
```

Comment:
Shouldn't need to track both circular and linear, because linear = not circular

Reply:
here I would see if the topology is linear and if not  especified as linear, assume circular given that the other option is circular and if not informed it assumes linear. This can change if more DNA topologies are added to SBOL in future but is very unlikely in the short term. So what do you think about assesing linearity and if not assume that is circular?

Reply:
The specification currently says that if topology is not circular, then one can assume linear.
I would thus recommend using that.

Reply:
now assuming linear if not circular